File: Data Structures & Algorithms/climbing-stairs/submission-0.java
class Solution {
    
    public int climbStartRec(int n, int[] dp) {
        if (n == 0) return 1;
        if (n < 0) return 0;
        if (dp[n] != -1) return dp[n];

        dp[n] = climbStartRec(n - 1, dp) + climbStartRec(n - 2, dp);

        return dp[n];
    }

    public int climbStairs(int n) {
        int[] dp = new int [n + 1];
        for(int i = 0; i < n + 1; i++) {
            dp[i] = -1;
        }

        return climbStartRec(n, dp);
    }
}

File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector <int> dict(26);

        for (char c : s) {
            dict[c - 'a']++;
        }

        for (char c : t) {
            dict[c - 'a']--;
        }

        for (int i = 0; i < 26; i++)
            if (dict[i] != 0)
                return false;

        return true;
    }
};
